lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -291,18 +287,23 @@ getTokenTransfer = label "getTokenTransfer" $ do
         value <- Coin <$> getWord64be
         return $ TxOut addr value
 
-    getAddress = do
-        headerByte <- getWord8
-        let kind = kindValue headerByte
-        let _discrimination = discriminationValue headerByte
-        case kind of
-            -- Single Address
-            0x3 -> Address <$> getByteString 32
-            0x4 -> error "unimplemented group address decoder"
-            0x5 -> error "unimplemented account address decoder"
-            0x6 -> error "unimplemented multisig address decoder"
-            other -> fail $ "Invalid address type: " ++ show other
 
+
+getAddress :: Get Address
+getAddress = do
+    -- We use 'lookAhead' to not consume the header, and let it
+    -- be included in the underlying Address ByteString.
+    headerByte <- label "address header" . lookAhead $ getWord8
+    let kind = kindValue headerByte
+    let _discrimination = discriminationValue headerByte
+    case kind of
+        -- Single Address
+        0x3 -> Address <$> getByteString 33
+        0x4 -> error "unimplemented group address decoder"
```

Comment:
How big of an effort is it to add support for group address? I expect it to be only `Address <$> getByteString 65` ?

Reply:
I suppose that will just work…

We won't be able to discover them in `isOurs` though as `keyToAddress` only can create one type of addresses (now single).

Reply:
True although the concern is slightly different here. BIP-44 for group addresses is out of the question for now. Support for delegation and delegation addresses will come later.